interview-questions.md

```diff
@@ -8,62 +8,61 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is iteration?
 
-  Your answer:
-
-  Researched answer:
-
+  Your answer:Iteration is number of execution in a loop until a certain condition met.
 
+  Researched answer:Iteration is reapting a certain steps or instruction until a ceratin condition is met.
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer:the predefined parameters taken by map higher order function are 
+  value ,index and array.
 
-  Researched answer:
+  Researched answer:Generally all built in higher order methods take three parameters and the y are value,index and an array.
 
 
 
 3. What is object destructuring?
 
-  Your answer:
+  Your answer:object destructuring is assigning the key values of an object property as a variable and assign to the object
 
-  Researched answer:
+  Researched answer:Destructuring is a programming  syntax that allows to unpack the elements of an array or an object and allow as to print a specific element by assigning the propery as a variable.
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer:All methods are functions but not all functions are methods
 
-  Researched answer:
+  Researched answer:All methods are functions but not all functions are methods.methods are always associated with the calling object but functions are not ,they can be invoked just by calling their names.
 
 
 
 5. What is the difference between a class and an object?
 
-  Your answer:
+  Your answer:Class is a blue print to factor objects where as objects are the tanjible instances created from the blue print
 
-  Researched answer:
+  Researched answer:The difference between class and object are ,classes are the syntactical sugars or the blue print to create a number of objects that may have different properties .
```

Comment:
Yes classes allow us to make infinite copies of a predefined object for use in development or production.